sdk/communication/Azure.Communication.Common/src/CommunicationUserIdentifier.cs

```diff
@@ -27,10 +30,26 @@ public class CommunicationUserIdentifier : CommunicationIdentifier
         /// <exception cref="System.ArgumentException">
         /// Thrown when the <paramref name="id"/> is empty.
         /// </exception>
-        public CommunicationUserIdentifier(string id)
+        public CommunicationUserIdentifier(string id) : this(id, CommunicationCloudEnvironment.Public)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="CommunicationUserIdentifier"/>.
+        /// </summary>
+        /// <param name="id">Id of the communication user.</param>
+        /// <exception cref="System.ArgumentNullException">
+        /// Thrown when the <paramref name="id"/> is null.
+        /// </exception>
+        /// <exception cref="System.ArgumentException">
+        /// Thrown when the <paramref name="id"/> is empty.
+        /// </exception>
+        /// <param name="cloud">The cloud that the ACS user belongs to.</param>
+        public CommunicationUserIdentifier(string id, CommunicationCloudEnvironment cloud)
         {
             Argument.AssertNotNullOrEmpty(id, nameof(id));
```

Comment:
Do you allow to have null for `cloud` or should there also be the same verification as with `id`?

Reply:
Good point. Cloud shouldn't be null. Will update.

Reply:
I removed the constructor altogether. Cloud property is now set in original constructor using dedicated private method.